Placeholders: Make dashed style a mixin, and apply across.

jasmussen opened the pull request:
## What?

As placeholders in the dashed-outline style have received polish in the last week, an opacity differential has stood out:

<img width="728" alt="before" src="https://user-images.githubusercontent.com/1204802/186138299-7e5da06c-6b5b-4246-91c0-72447207df85.png">

This PR makes the styles the same across, and extracts the particular dash-style into a mixin so it has a better chance to stay in sync across changes:

<img width="748" alt="after" src="https://user-images.githubusercontent.com/1204802/186138397-e0fad7f9-4295-4d25-b2d1-83fe9dea233a.png">

## Testing Instructions

Test placeholders with the new dashed style, Image, Featured Image, Site Logo, and Columns. Dashed styles should look the same.

I noticed a regression with Group, separate from this PR, which I'll follow up on.

jasmussen committed:
Placeholders: Make dashed style a mixin, and apply across.

jameskoster commented on packages/block-library/src/group/editor.scss:
```diff
@@ -39,12 +39,12 @@
 		pointer-events: none;
 		min-height: $grid-unit-60 - $border-width - $border-width;
+
+		// Dashed placeholder style.
```
It's a real nitpick, but do we really need the comment? If it's a general codestyle thing then fine, but otherwise it's kind of duplicative considering it includes the exact same words as the mixin name :D

jasmussen replied:
Nitpicks like these are my favorite 👌

jameskoster commented on packages/base-styles/_mixins.scss:
```diff
@@ -202,6 +202,11 @@
 	}
 }
 
+@mixin dashed-placeholder-style() {
```
Might we consider `placeholder-style` as the name instead, just incase we decide to change the border style in the future? 🤔

jasmussen committed:
Rename to placeholder-style and remove comments.

jasmussen commented:
Thank you for the review! Rebased, renamed mixin, removed comments. Still looks good, here's an image and 2 columns:
<img width="827" alt="Screenshot 2022-08-26 at 10 38 33" src="https://user-images.githubusercontent.com/1204802/186860886-aefa721b-3673-4823-95d0-cb6e4b8f92ca.png">

jameskoster reviewed (approved):
Bingo!

Mamaduka commented:
Hi, @jasmussen

I  think this enhancement got regressed after #44190. The dashed border style is missing on the current trunk.

jasmussen commented:
Visually it was intentional to move forward with the borderless version in 44190 to reduce noise as the placeholders become increasingly useful in patterns. That new style is still in need of some iteration, see #44460, and as it stands, 6.1 should ship with the bordered version. Or is there a different aspect to the regression I was missing?

Mamaduka commented:
Sorry, I wasn't clear.

The #44190 also removed borders for Group/Columns appender placeholders. So now, unselected empty Group or Columns isn't visible on the canvas.

## Screenshot

![CleanShot 2022-10-05 at 11 15 29](https://user-images.githubusercontent.com/240569/194002585-ca73e5e9-8a89-460e-961d-a29c6e8e57eb.png)

jasmussen commented:
Oh that's no good! I'll take a look, thanks.
